js/tji-datasets-explore.js

```diff
@@ -44,20 +44,16 @@ TJIGroupByBarChart.prototype.create = function(data) {
   var that = this;
   var grouped = this.get_group_counts(data);
 
-  // 'create' is only run with the full data set.
-  // So we store our color mapping (e.g. male = blue, female = red)
-  // so that if the user ends up filtering some of them out, we can
-  // prevent chartjs from altering the label-color pairings.
-  this.color_mapping = {};
-  _.each(grouped.keys, function(k, idx) {
-    that.color_mapping[k] = that.color_palette[idx % that.color_palette.length];
-  });
-
-  // Apply color mapping
-  var colors = _.map(grouped.keys, function(k) {
-    return that.color_mapping[k]
+  var colors = _.map(grouped.keys, function(k, i) {
+    return that.color_palette[i % that.color_palette.length];
   })
 
+  // 'create' is only run with the full data set.
+  // We store the 'full' set of unique groupBy values,
```

Comment:
We store the 'full' set of unique groupings
since we want to show every unique value in the legend,
even if the user filters some of the groupings out later.
Ex. For 'Race', we want to always show Black, Hispanic, White, Other, ... 
in the legend even if the 'Black' filter is not selected.